docs/HARDWARE-BRINGUP.md

```diff
@@ -50,6 +50,108 @@ One word of caution: a lot of ARM board (like the ever popular Raspberry Pi) act
+
+#### 4.1.2. Hardware discovery via device tree
+
+For device tree the process is effectively the same, although unlike ACPI tables, in most situation you can control how device tree looks to the Linux kernel directly. The trick with device trees and Linux kernel is that the two require a great deal of coupling and it is NOT uncommon to have device trees that would work with kernel version X and break with kernel version Y. This is exactly why most reliable device trees still come out of the linux kernel build (under `arch/X/boot/dts/` folder). EVE prefers all of its device trees to be available in the source form (to facilitate rebasing on newer kernels) and we try to use as much of the kernel own's device tree source as possible (by moving all the custom tweaks to device trees into standalone DTS files that include the bulk of the definitions from the kernel and simply tweak the rest - see [this as a good example](https://github.com/lf-edge/eve/blob/6.4.0/pkg/new-kernel/patches-5.11.x/0021-Add-uno-220-dts.patch)). Still there are times when you have to deal with vendor provided device tree binary blobs (either given to you as a file OR given to the Linux kernel by the UEFI BIOS). At that point you deal with them as you would deal with raw ACPI tables but you would use [dtc](https://pkgs.alpinelinux.org/package/edge/main/x86/dtc) tool to disassemble them via `dtc -I fs -O dts /sys/firmware/device-tree/base` and then manually inspect their elements.
+
+The Linux kernel view of the device hierarchy created based on the information presented in a device tree is available under `/sys/devices/platform`.
+
+### 4.2. Bus discovery
+
+Having a manifest of all the hardware present on the system is only as useful as CPU's ability to talk to all that hardware. Some of the hardware can be driven directly by memory mapped registers and DMA, other require buses to be available. A bus itself is a kind of a hardware device (at least the end of the bus that connects it to the host CPU) and Linux kernel tracks all the buses it knows about under `/sys/bus/`. Before you can move to enabling a driver, make sure that the required bus is visible under `/sus/bus/BUS-NAME`. The most interesting subfolder under that is `/sus/bus/BUS-NAME/devices`. This is Linux kernel's idea of what devices sit on that bus. Sophisticated buses (like USB) allow for dynamic device enumeration and discovery so that `/sus/bus/usb/devices` is maintained up-to-date by the bus driver itself. Simpler buses (like I2C) don't allow for device discovery. The device sitting on such a bus has to either be described in an ACPI table or device tree entry OR it has to be [manually instantiated](https://www.kernel.org/doc/html/latest/i2c/instantiating-devices.html). For example, the following will instantiate device with the name `eeprom` at I2C address `0x50` on an I2C bus #3:
```

Comment:
I think you meant "is visible under `/sys/bus/BUS-NAME`" and not `/sus/`

Reply:
Actually, that typo appears several times.

Reply:
The magic of cut-n-paste ;-) well spotted -- fixed!